Tools/cdm-types/cdm-types.ts

```diff
@@ -112,11 +117,21 @@ export interface TypeAttribute
     defaultValue?: any;
 }
 
+let isTypeAttribute = (object): object is TypeAttribute =>
+{
+    return "name" in object;
```

Comment:
with recent changes, all attributes must have names. the only way to tell the diff would be to look for an 'entity' property on the entityAttribute. you could look for 'dataType' but, alas, that is optional.